### PR TITLE
PR: Update psutil dependency constraint and add checks for dependency integrity

### PR DESCRIPTION
**Pull request by goanpeca**
### Issue(s) Resolved

Fixes https://github.com/spyder-ide/spyder/issues/11497

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by goanpeca** [comment]
@ccordoba12 we need to update the conda recipe for the new spyder. Should I open a separate issue to keep track of that @ccordoba12 ?

**Comment by ccordoba12** [comment]
> Should I open a separate issue to keep track of that?

Please open a pull request in our [feedstock](https://github.com/conda-forge/spyder-feedstock) to generate a new build of 4.0.1 that includes it.

**Comment by goanpeca** [comment]
https://github.com/conda-forge/spyder-feedstock/pull/74

**Comment by ccordoba12** [comment]
@goanpeca, please don't forget to update our `setup.py` with this new constraint, so we can merge this one.

**Comment by goanpeca** [comment]
True I forgot, let me fix it!

**Comment by goanpeca** [comment]
I will add a test to check that our requirements, the dependencies file and the setup.py are in sync otherwise we might have issues in the future.

**Comment by ccordoba12** [comment]
> I will add a test to check that our requirements, the dependencies file and the setup.py are in sync otherwise we might have issues in the future.

Sounds good, thanks!

**Comment by goanpeca** [comment]
@ccordoba12 in writing the test for dependency integrity I found a lot of inconsistencies, like, deps in the Spyder dialog where defined, but not in the setup.py etc...

We should now have everything in sync :-)

File: spyder/dependencies.py [hunk 1]
```diff
@@ -111,41 +107,31 @@
...
-    )
-
-
-DESCRIPTIONS += [
+     'required_version': NUMPYDOC_REQVER},
+    {'modname': "paramiko",
+     'package_name': "paramiko",
+     'features': _("Connect to remote kernels through SSH"),
+     'required_version': PARAMIKO_REQVER,
+     'display': sys.platform == 'nt'},
```
Comment (ccordoba12) [inline spyder/dependencies.py]: This needs to be `os.name == 'nt'`. This was an error on my side (I think).

Comment (goanpeca) [reply spyder/dependencies.py]: Will fix

Comment (goanpeca) [reply spyder/dependencies.py]: Fixed!

**Comment by goanpeca** [comment]
Ok please do not merge yet, I need to add an extra check for setup.py extras_require

**Comment by goanpeca** [comment]
This one is ready @ccordoba12 !

**Pull request merged**